src/scale.c

```diff
@@ -180,3 +182,33 @@ avifBool avifImageScale(avifImage * image,
 }
 
 #endif
+
+avifResult avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes)
+{
+    for (int c = AVIF_CHAN_Y; c <= AVIF_CHAN_A; ++c) {
+        if ((!(planes & AVIF_PLANES_YUV) && (c != AVIF_CHAN_A)) || (!(planes & AVIF_PLANES_A) && (c == AVIF_CHAN_A))) {
+            continue;
+        }
+        const uint32_t planeWidth = avifImagePlaneWidth(srcImage, c);
+        const uint32_t planeHeight = avifImagePlaneHeight(srcImage, c);
+        const uint8_t * srcRow = avifImagePlane(srcImage, c);
+        uint8_t * dstRow = avifImagePlane(dstImage, c);
+        const uint32_t srcRowBytes = avifImagePlaneRowBytes(srcImage, c);
+        const uint32_t dstRowBytes = avifImagePlaneRowBytes(dstImage, c);
+        if ((planeWidth != avifImagePlaneWidth(dstImage, c)) || (planeHeight != avifImagePlaneHeight(dstImage, c)) ||
+            (!srcRow != !dstRow) || (srcImage->depth != dstImage->depth)) {
+            return AVIF_RESULT_INVALID_ARGUMENT;
+        }
+        if (!srcRow) {
+            continue;
+        }
+
+        const size_t planeWidthBytes = planeWidth * (size_t)(avifImageUsesU16(srcImage) ? 2 : 1);
```

Comment:
Nit: Similarly the bytes-per-pixel expression is loop invariant and can be saved in a local variable outside the for loop.

Also in `avifImageCopy()` we use this form (left shift):

```
       uint32_t planeWidthBytes = avifImagePlaneWidth(dstImage, plane) << (dstImage->depth > 8);
```

Would be good to standardize on one form (either multiplication or left shift).